oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandler.java

```diff
@@ -298,22 +308,21 @@ public APIGatewayProxyResponseEvent authoriseRequestHandler(
         if (isJarValidationRequired && authRequest.getRequestObject() == null) {
             String errorMsg = "JAR required for client but request does not contain Request Object";
             LOG.warn(errorMsg);
-            var errorResponse =
-                    new AuthenticationErrorResponse(
-                            authRequest.getRedirectionURI(),
-                            new ErrorObject(ACCESS_DENIED_CODE, errorMsg),
-                            authRequest.getState(),
-                            authRequest.getResponseMode());
             if (client.getRedirectUrls().contains(authRequest.getRedirectionURI().toString())) {
                 LOG.warn("Redirecting");
-                return generateApiGatewayProxyResponse(
-                        302,
-                        "",
-                        Map.of(ResponseHeaders.LOCATION, errorResponse.toURI().toString()),
-                        null);
+                return generateErrorResponse(
+                        authRequest.getRedirectionURI(),
+                        authRequest.getState(),
+                        authRequest.getResponseMode(),
+                        new ErrorObject(ACCESS_DENIED_CODE, errorMsg),
+                        client.getClientID(),
+                        user);
             } else {
-                LOG.warn("Redirect URI not found in client registry");
-                return RedirectService.redirectToFrontendErrorPage(authFrontend.errorURI());
+                LOG.warn(
+                        "Redirect URI {} is invalid for client {}",
```

Comment:
Nitpick but in general we shouldn't need to insert the client ID into logs like this as it will be attached to every log